conformance_test.go

```diff
@@ -0,0 +1,106 @@
+/*
+Copyright 2022 The Dapr Authors
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+    http://www.apache.org/licenses/LICENSE-2.0
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package main
+
+import (
+	"os"
+	"testing"
+
+	confBindings "github.com/dapr/components-contrib/tests/conformance/bindings"
+	confPubsub "github.com/dapr/components-contrib/tests/conformance/pubsub"
+	confState "github.com/dapr/components-contrib/tests/conformance/state"
+	"github.com/dapr/components-contrib/tests/conformance/utils"
+
+	"github.com/mitchellh/mapstructure"
+	"gopkg.in/yaml.v3"
+
+	"github.com/dapr/dapr/pkg/components/bindings"
+	"github.com/dapr/dapr/pkg/components/pubsub"
+	"github.com/dapr/dapr/pkg/components/state"
+
+	"github.com/dapr/kit/logger"
+
+	"github.com/stretchr/testify/require"
+)
+
+var l = logger.NewLogger("dapr-conformance-tests")
+
+func TestPluggableConformance(t *testing.T) {
+	socket := "/tmp/socket.sock"
+	if soc, ok := os.LookupEnv("INPUT_SOCKET"); ok {
+		socket = soc
+	}
+
+	var componentMetadata map[string]string
+	if metadata, ok := os.LookupEnv("INPUT_METADATA"); ok {
+		require.NoError(t, yaml.Unmarshal([]byte(metadata), &componentMetadata))
```

Comment:
I would suggest not doing activity and assertion on the same line, for readability.

```go
err = yaml.Unmarshal([]byte(metadata), &componentMetadata)
require.NoError(t, err)
```

Reply:
Also below multiple times

Reply:
I would agree if the line is too long that prevents you to understand what it is doing, but these lines are just a single and well-known purposes for parsing data,

I think it's worth opening it in the IDE (if you haven't already) to make sure it's a problem, I think it's OK to keep it that way. If you're still reluctant about it, I can change it, no worries. But particularly I prefer that way, if you don't have any other stronger reason to change it.

Reply:
For me, it's just more readable to split the actual activity and assertion, as a principle, but it's not a deal-breaker.